src/ApiService/ApiService/Program.cs

```diff
@@ -12,43 +12,66 @@
 using Azure.Identity;
 using Microsoft.ApplicationInsights.DependencyCollector;
 using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
 using Microsoft.Azure.Functions.Worker.Middleware;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
 using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
 using Microsoft.FeatureManagement;
 using Microsoft.Graph;
 using Microsoft.OneFuzz.Service.OneFuzzLib.Orm;
-
 namespace Microsoft.OneFuzz.Service;
 
 public class Program {
+
+    /// <summary>
+    /// 
+    /// </summary>
     public class LoggingMiddleware : IFunctionsWorkerMiddleware {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="context"></param>
+        /// <param name="next"></param>
+        /// <returns></returns>
         public async Async.Task Invoke(FunctionContext context, FunctionExecutionDelegate next) {
-            var log = (ILogTracerInternal?)context.InstanceServices.GetService<ILogTracer>();
-            if (log is not null) {
-                //TODO
-                //if correlation ID is available in HTTP request
-                //if correlation ID is available in Queue message
-                //log.ReplaceCorrelationId(Guid from request)
-
-                log.ReplaceCorrelationId(Guid.NewGuid());
-                log.AddTags(new[] {
-                    ("InvocationId", context.InvocationId.ToString())
-                });
+            //https://learn.microsoft.com/en-us/azure/azure-monitor/app/custom-operations-tracking#applicationinsights-operations-vs-systemdiagnosticsactivity
+            using var activity = OneFuzzLogger.Activity;
+            _ = activity.Start();
+            string correlationId = Guid.NewGuid().ToString();
+
+            if (await context.GetHttpRequestDataAsync() is HttpRequestData requestData) {
+                //if header has 1f-CorrelationId then use that
+                //otherwise check if message can be deserialized to {"correlationId": "SOME-GUID"}, then use that
+
+                if (requestData.Headers.TryGetValues("CorrelationId", out var values1f)) {
+                    correlationId = values1f.First();
+                } else if (requestData.Headers.TryGetValues("x-correlationId", out var values)) {
```

Comment:
I think a more standard format for HTTP headers would look like "X-Correlation-ID"?

Reply:
yeah 

https://en.wikipedia.org/wiki/List_of_HTTP_header_fields

X-Correlation-ID,[[43]](https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#cite_note-44) 
Correlation-ID[[44]](https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#cite_note-45)